### PR TITLE
vim-patch:8.1.2227,8.2.{315,928,1007,1052,1121,1580,2221,2229,2231,2232}

### PR DESCRIPTION
**Pull request by janlazo**

### CODE REVIEW

File: src/nvim/ex_getln.c [hunk 1]
```diff
@@ -1082,6 +1082,7 @@ static int command_line_execute(VimState *state, int key)
     if (s->c == K_DOWN && ccline.cmdpos > 0
         && ccline.cmdbuff[ccline.cmdpos - 1] == '.') {
       s->c = (int)p_wc;
```
Comment (bfredl) [inline src/nvim/ex_getln.c:1084]: Hmm, `Ctrl_Z` should always work in neovim without needing to fudge `KeyTyped`.

Comment (janlazo) [reply src/nvim/ex_getln.c:1084]: Ok. I'll test in a separate commit.

File: src/nvim/testdir/test_cmdline.vim [hunk 2]
```diff
@@ -51,6 +51,19 @@ func Test_complete_wildmenu()
   call feedkeys(":e Xdir1/\<Tab>\<Down>\<Up>\<Right>\<CR>", 'tx')
   call assert_equal('testfile1', getline(1))
 
+  +  " <C-J>/<C-K> mappings to go up/down directories when 'wildcharm' is
+  " different than 'wildchar'.
+  set wildcharm=<C-Z>
```
Comment (janlazo) [inline src/nvim/testdir/test_cmdline.vim:56]: `wildcharm` option is set to non-zero so https://github.com/neovim/neovim/commit/bed9839f46047abd81f7c9b56936ba177829221c won't work here.

Comment (bfredl) [reply src/nvim/testdir/test_cmdline.vim:56]: bed9839 should always work as it is enabled unconditionally.

Comment (janlazo) [reply src/nvim/testdir/test_cmdline.vim:56]: `wildchar` is also non-zero by default.

Comment (bfredl) [reply src/nvim/testdir/test_cmdline.vim:56]: still, bed9839 keeps working.

File: src/nvim/ex_getln.c (hunk 1 above)
Comment (janlazo) [reply src/nvim/ex_getln.c:1084]: It doesn't work.

File: src/nvim/testdir/test_cmdline.vim (hunk 2 above)
Comment (janlazo) [reply src/nvim/testdir/test_cmdline.vim:56]: The test fails without the C code changes.

**Pull request merged**